src/rosdiscover/observer/observer.py

```diff
@@ -45,17 +48,69 @@ def for_container(cls,
         rsw = roswire.ROSWire()
         app = rsw.app(config.image, config.sources)
         instance = app.attach(container, require_description=True)
+        observer = cls._build_observer(app, config, instance)
+        observer._attached = None
+        return observer
+
+    @classmethod
+    def _build_observer(cls, app, config, instance):
         if app.description.distribution.ros == ROSVersion.ROS1:
             from .ros1 import ROS1Observer
             return ROS1Observer(instance, config)
         else:
             from .ros2 import ROS2Observer
             return ROS2Observer(instance, config)
 
+    @classmethod
+    def for_image(cls, config: 'Config') -> 'Observer':
+        """Constructs an observer by starting an instance of an image
+
+        Parameters
+        ----------
+        config: Config
+            A description of the configuration used by the running container.
+            Will contain the image name, sources, and environment variables to
+            run
+
+        Returns
+        -------
+        Observer
+            An observer that is appropriate for the kind of ROS system that is running in the
+            container.
+        """
+        rsw = roswire.ROSWire()
+        app = rsw.app(config.image, config.sources)
+        instance = app.launch(environment=config.environment)
+        observer = cls._build_observer(app, config, instance)
+        observer._attached = False
+
+        logger.debug("Starting the container")
+        cmd = "/startup-vnc.sh"
```

Comment:
this seems hardcoded to TheRobotCooperative images? We should probably make this a parameter or create an issue.